### PR TITLE
comment for Expression

### PR DESCRIPTION
**Pull request by chenzhitong**

### CODE REVIEW

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/AnonymousObjectCreationExpression.cs [hunk 1]
```diff
@@ -18,6 +18,21 @@ namespace Neo.Compiler;
 
 partial class MethodConvert
 {
+    /// <summary>
+    /// Methods for converting the creation of an object of an anonymous type into a series of instructions.
+    /// </summary>
+    /// <param name="model">The semantic model providing context and information about the anonymous object creation.</param>
+    /// <param name="expression">The syntax representation of the  anonymous object creation statement being converted.</param>
+    /// <remarks>
+    /// Anonymous types provide a convenient way to encapsulate a set of read-only properties into a single object without having to explicitly define a type first. The type name is generated by the compiler and is not available at the source code level. The type of each property is inferred by the compiler.
```
Comment (Jim8y) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/AnonymousObjectCreationExpression.cs:27]: too long, break into two lines please

**Comment by Jim8y** [comment]
Please keep going with this excellent work. More comments need to be added.

**Comment by Jim8y** [comment]
Please check your newly added comments, missing examples and what you mean IL instruction here?

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/InvocationExpression.cs [hunk 2]
```diff
@@ -73,6 +95,15 @@ private void ConvertMethodInvocationExpression(SemanticModel model, IMethodSymbo
...
     }
 
+    /// <summary>
+    /// Convert the delegate invocation expression to OpCodes.
+    /// </summary>
+    /// <param name="model">The semantic model providing context and information about delegate invocation expression.</param>
+    /// <param name="expression">The syntax representation of the delegate invocation expression statement being converted.</param>
+    /// <param name="arguments">Arguments of the delegate</param>
+    /// <example>
+    /// TODO
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/InvocationExpression.cs:105]: TODO

File: src/Neo.Compiler.CSharp/MethodConvert/Statement/BlockStatement.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@ namespace Neo.Compiler
     partial class MethodConvert
     {
         /// <summary>
-        /// Converts a block statement to a sequence of instructions. This method is used for parsing
+        /// Converts a block statement to OpCodes. This method is used for parsing
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert/Statement/BlockStatement.cs:21]: Instructions it's better isn't it? OpCode is the kind of instruction, but instruction have also the data

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/Expression.cs [hunk 4]
```diff
@@ -96,67 +102,113 @@ private void ConvertExpression(SemanticModel model, ExpressionSyntax syntax, Syn
...
+            //Converts a switch  expression.
+            //Example: day switch { 1 => "Monday", 2 => "Tuesday", 3 => "Wednesday", 4 => "Thursday", 5 => "Friday", 6 => "Saturday", 7 => "Sunday", _ => "Unknown" };
             case SwitchExpressionSyntax expression:
                 ConvertSwitchExpression(model, expression);
                 break;
+            //Converts a base  expression.
+            //Example: public A() : base() { }
             case BaseExpressionSyntax:
+            //Converts a extended methods
+            //Examples: UInt160.Zero.ToAddress();
```
Comment (Jim8y) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/Expression.cs:200]: please double check this

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/IdentifierNameExpression.cs [hunk 5]
```diff
@@ -20,6 +20,27 @@ namespace Neo.Compiler;
...
+    /// <param name="expression">The syntax representation of the identifier name expression statement being converted.</param>
+    /// <exception cref="CompilationException">Unsupported symbols will result in a compilation exception.</exception>
+    /// <example>
+    /// Processing of the identifier "param" goes to the "IParameterSymbol parameter" branch of the code;
+    /// processing of the identifier "temp" goes to the "ILocalSymbol local" branch of the code.
+    /// Unused identifier "param2" will not be processed.
+    /// <code>
+    /// public static void MyMethod(int param, int param2)
+    /// {
+    ///     int temp = int.MaxValue;
```
Comment (Jim8y) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/IdentifierNameExpression.cs:36]: which line exactly is the identifiernameexpression?

**Review by Jim8y — APPROVED** [review]

File: src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs [hunk 6]
```diff
@@ -22,6 +22,23 @@ namespace Neo.Compiler;
 partial class MethodConvert
 {
 
+    /// <summary>
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert/Expression/AssignmentExpression.SimpleAssignment.cs:25]:
```suggestion
    /// <summary>
```

File: src/Neo.Compiler.CSharp/MethodConvert/Statement/BlockStatement.cs (hunk 3 above)
Comment (shargon) [reply src/Neo.Compiler.CSharp/MethodConvert/Statement/BlockStatement.cs:21]: @Jim8y what do you think?

**Review by Jim8y — APPROVED** [review]

File: src/Neo.Compiler.CSharp/MethodConvert/Statement/BlockStatement.cs (hunk 3 above)
Comment (shargon) [reply src/Neo.Compiler.CSharp/MethodConvert/Statement/BlockStatement.cs:21]: gif( cri cri )

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert/Statement/BlockStatement.cs:21]: @shargon I dont really have opinion on it, you can update it if you think it is right.  My bad for merging it, I thought you approved it as i checked it on phone and it shows you reviewed.

**Pull request merged**